.github/workflows/macos.yaml

```diff
@@ -0,0 +1,38 @@
+name: Test build on MacOS
+
+on:
+  push:
+    branches: [ "master", "auto" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - name: Install prerequisite packages
+      run: |
+        brew install \
+          automake coreutils cppunit gawk \
+          gnu-getopt gnu-sed grep \
+          make openldap cyrus-sasl
+    - name: Checkout sources
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 5
+    - name: Prepare build
+      run: |
+        export PKG_CONFIG_PATH="`brew list openldap | grep ldap.pc | head -1 | xargs dirname`:`brew list cyrus-sasl | grep libsasl2.pc | head -1 | xargs dirname`";
+        export CFLAGS="-Wno-compound-token-split-by-macro ";
+        export CPPFLAGS="`pkg-config --cflags ldap libsasl2`";
+        export LDFLAGS="`pkg-config --libs ldap libsasl2`";
```

Comment:
It would be best to update `configure.ac` to do these.

Reply:
ldap-related configure.ac is a mess :\
Note that this is only relevant on macos, because:

Reply:
> ldap-related configure.ac is a mess

IMO, that assertion is not a valid excuse to create _new mess_ in GitHub Actions configuration.

Please note that it is perfectly fine to (modify configure.ac slightly) to make GitHub Actions work on macos _without_ enabling LDAP features (on macos). We do not have to spend a lot of resources to enable LDAP testing on macos in this PR. Moreover, the more custom code we need to add to enable such testing, the higher the probability that we should not add that custom code at all (and should effectively disable that testing instead).